### ISSUE TITLE
binding for etoolbox

### ISSUE THREAD

**Issue by kohlhase** [brucemiller/LaTeXML#478]
I find myself using the `etoolbox` package more and more. It provides nice programming primitives for LaTeX. And at the same time, I like supporting LaTeXML by just `RawTeX`ing parts of the LaTeX packges (at least for roughing things out). Currently, that means that I cannot use `etoolbox` primitives. 

I must confess that I have not tried just inputting `etoolbox.sty` and letting LaTeXML figure things out, but I think that in the end this will not be efficient. 

Therefore, ... a binding would be nice to have.

**Comment by dginev** [comment]
Btw, I am now of the opinion that before we start writing Perl bindings we should understand the failures with the native TeX interpretation and improve LaTeXML to facilitate a clean load (and use) of etoolbox.sty.

Once that works, we can optimize for performance by rewriting the slowest parts in Perl. I would advertise this as a general approach to new bindings we consider supporting, so that we get ever closer to the magical 1.0 release (where we have indistinguishably close coverage to classic TeX)

**Comment by brucemiller** [comment]
Good point.  In fact, I opened a somewhat related issue #622 more related to coverage than performance.

As far as etoolbox is concerned, since it is mainly 'programming', it might just work. The only way to determine is for somebody to write a somewhat comprehensive test document that exercises all or most of the macros, and then compare pdflatex to latexml with --includestyles.  If the comparison is good, we can create a dummy binding that simply loads etoolbox.sty.

Of course, the test above indicates it won't yet go.  I don't seem to have etoolbox.sty on my work system, but etex.sty fails due to a random latex-internal counter (\insc@unt).

**Comment by dginev** [comment]
> As far as etoolbox is concerned, since it is mainly 'programming', it might just work.

From today's test I randomly found out, it doesn't currently:
```
Attempt to pop last locked stack frame at /usr/share/texlive/texmf-dist/tex/latex/etoolbox/etoolbox.sty; line 779 col 18 In Core::Stomach[@0x3c490f8] at /usr/share/texlive/texmf-dist/tex/latex/etoolbox/etoolbox.sty; line 779 col 18 <= Core::Definition::Primitive[End] <= Core::Stomach[@0x3c490f8] <= Core::Gullet[@0x3cabc58] <= Core::Definition::Constructor[\Requir... <= Core::Stomach[@0x3c490f8] <= Core::Gullet[@0x3cabc58] <= Core::Definition::Constructor[\usepac... <= Core::Stomach[@0x3c490f8]
--
```
source: https://corpora.mathweb.org/corpus/%2Fdata%2Farxiv_1802%2F/raw_tex_to_html/fatal/unexpected/%3Cendgroup%3E

And apparently near 10% of tasks used it, so I'm definitely taking a look at the native interpretation, could lead to all kinds of progress.

**Comment by dginev** [comment]
Solved by merging #978
